CharlieBackend.AdminPanel/HtmlHelpers/CalendarHelper.cs

```diff
@@ -0,0 +1,200 @@
+﻿using CharlieBackend.AdminPanel.Models.Calendar;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Encodings.Web;
+
+namespace CharlieBackend.AdminPanel.HtmlHelpers
+{
+    public static class CalendarHelper
+    {
+
+        public static HtmlString CalendarBodyHtml(this IHtmlHelper html, CalendarViewModel calendar)
+        {
+            DateTime start, finish;
+
+            var eventOccurencesFiltered = calendar.ScheduledEvents.Select(x => calendar.EventOccurences.First(y => y.Id == x.EventOccuranceId)).ToList();
+
+            if (calendar.ScheduledEventFilter.FinishDate.HasValue)
+            {
+                finish = calendar.ScheduledEventFilter.FinishDate.Value;
+            }
+            else
+            {
+                DateTime latestFinish = eventOccurencesFiltered.Max(x => x.EventFinish);
+                DateTime latestStart = eventOccurencesFiltered.Max(x => x.EventStart);
+
+                finish = latestFinish > latestStart ? latestFinish : latestStart;
+            }
+
+            if (calendar.ScheduledEventFilter.StartDate.HasValue)
+            {
+                start = calendar.ScheduledEventFilter.StartDate.Value;
+            }
+            else
+            {
+                DateTime earliestFinish = eventOccurencesFiltered.Min(x => x.EventFinish);
+                DateTime earliestStart = eventOccurencesFiltered.Min(x => x.EventStart);
+
+                start = earliestFinish > earliestStart ? earliestStart : earliestFinish;
+            }
+
+            double daysCount = (finish - start).TotalDays;
+            double rowCount = daysCount / 7;
+
+            if (rowCount - (int)rowCount != 0)
+                rowCount = (int)rowCount + 1;
+
+            if (finish.DayOfWeek != DayOfWeek.Saturday)
+                daysCount += 7 - (int)finish.DayOfWeek;
```

Comment:
https://github.com/ita-social-projects/WhatBackend/pull/464#discussion_r591616791
Please update all if statements to enclose logic inside `{ }` blocks

Reply:
done https://github.com/ita-social-projects/WhatBackend/pull/464/commits/11bf7a3896bb7502a80fa15d8b3d9535eb08d1ec